dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -1,820 +1,440 @@
-  }
-);
-
-// Create, update
-router.(post|put)(
-  {
-    path: '/api/resource',
-    validate: {
-      body: schema.object({
-        name: schema.string(),
-        config: schema.object({...}),
-      }),
-    },
-  },
-  async (context, request, response) => {
-    // Implementation...
-    return response.ok({ body: completeResourceState });
-  }
-);
-
-// Delete
-router.delete(
-  {
-    path: '/api/resource/{id}',
-    validate: {
-      params: schema.object({
-        id: schema.string(),
-      }),
-    },
-  },
-  async (context, request, response) => {
-    // Implementation...
-    return response.ok();
-  }
-);
-
-// List many resources
-router.get(
-  {
-    path: '/api/resource',
-    validate: {
-      query: schema.object({
-        page: schema.maybe(schema.number()),
-        perPage: schema.maybe(schema.number()),
-      }),
-    },
-  },
-  async (context, request, response) => {
-    // Implementation...
-    return response.ok({
-      body: {
-        items: resources,
-        total: totalCount,
-        page: currentPage,
-        perPage: itemsPerPage
-      }
-    });
-  }
-);
+router.get({ path: '/api/foos' ... }, async () => {...})
 ```
 
-#### API imposed challenges
-
-API design decisions can create real challenges for Terraform resource implementation. The Elasticsearch [Index Settings API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-put-settings) (\`PUT /\{index\}/\_settings\`) treats static settings (which require index recreation) and dynamic settings (which can be updated in place) identically. This creates a complex situation forcing the implementation to handle all settings together because the API doesn't offer a way to distinguish them upfront.
+You can make an HTTP API public by changing the `access` setting to `public`:
 
-```bash
-{
-  "error": {
-    "reason": "Can't update non dynamic settings [[index.codec, index.number_of_shards]] for open indices"
-}
+```typescript
+router.get({ path: '/api/foos', options: { access: 'public' } }, async () => {...})
 ```
 
-[Index settings update bug](https://github.com/elastic/terraform-provider-elasticstack/issues/52)
+See the principle regarding [commitment](#commitment) before making APIs public!
 
-A better API design would separate these concerns with different endpoints or provide metadata about which settings require recreation.
+## Structure and conventions
 
-### Handle resource references and dependencies
+The following are a quick reference for how to structure your HTTP APIs.
 
-Terraform configurations frequently define multiple resources that depend on each other. Your API design needs to account for these interdependencies in a way that enables Terraform's declarative model to work smoothly.
+### RESTful-like APIs
 
-#### Use stable, predictable resource identifiers
+Kibana HTTP APIs do not need to implement the full RESTful specification. We aim for a more pragmatic but still user-friendly approach.
```

Comment:
If we do keep the point about `RESTful-like` design, then we need to give a bit more detail on when and how to be pragmatic and retain user-friendliness. If not and if we don't have anything concrete in mind, then maybe rephrasing it as: If in doubt, default to ESTful API design.

Reply:
Yeah, the point I was trying to raise is a bit nuanced. Something like: we follow a subset of RESTful practices (as opposed to full RESTful) instead of RPC or anything else. But I think this is all implied from the stuff below. I'll remove this for now!